src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,454 @@
+import accessManager = require("common/shell/accessManager");
+import eventsCollector = require("common/eventsCollector");
+import jsonUtil = require("common/jsonUtil");
+import getDatabaseSettingsCommand = require("commands/database/settings/getDatabaseSettingsCommand");
+import saveDatabaseSettingsCommand = require("commands/database/settings/saveDatabaseSettingsCommand");
+import virtualGridController = require("widgets/virtualGrid/virtualGridController");
+import hyperlinkColumn = require("widgets/virtualGrid/columns/hyperlinkColumn");
+import textColumn = require("../../../widgets/virtualGrid/columns/textColumn");
+import saveDatabaseSettingsConfirm = require("./saveDatabaseSettingsConfirm");
+import models = require("../../../models/database/settings/databaseSettingsModels");
+
+type viewModeType = "summaryMode" | "editMode";
+type entriesGroupType = "allEntries" | "databaseEntries" | "customizedDatabaseEntries";
+
+class databaseSettings extends viewModelBase {
+    
+    allEntries = ko.observableArray<models.settingsEntry>([]);
+    groupToShow = ko.observable<entriesGroupType>("allEntries");
+    isAnyMatchingEntries: KnockoutComputed<boolean>;
+    
+    allCategoryNames: KnockoutComputed<Array<string>>;
+    selectedCategory = ko.observable<string>();
+
+    private categoriesGridController = ko.observable<virtualGridController<string>>();
+    private summaryGridController = ko.observable<virtualGridController<models.settingsEntry>>();
+
+    viewMode = ko.observable<viewModeType>("summaryMode");
+    editModeHasBeenEntered = false;
+
+    isSaveEnabled: KnockoutComputed<boolean>;
+
+    spinners = {
+        refresh: ko.observable<boolean>(false),
+        save: ko.observable<boolean>(false)
+    };
+
+    isForbidden = ko.observable<boolean>(false);
+    filterKeys = ko.observable<string>("");
+
+    constructor() {
+        super();
+        
+        this.bindToCurrentInstance("save", "refresh", "switchToEditMode", "exitEditMode", "setGroupToShow");
+        this.initializeObservables();
+    }
+
+    private initializeObservables() {
+
+        this.initializeDirtyFlag();
+        
+        this.isSaveEnabled = ko.pureComputed(() => {
+            const isDirty = this.dirtyFlag().isDirty();
+            return isDirty && !this.spinners.save();
+        });
+
+        this.allCategoryNames = ko.pureComputed(() => {
+            return _.uniq(this.allEntries().map(entry => entry.data.Metadata.Category)).sort();
+        });
+
+        this.selectedCategory.subscribe(() => this.computeEntriesToShow());
+        this.groupToShow.subscribe(() => this.computeEntriesToShow());
+        this.filterKeys.throttle(500).subscribe(() => this.computeEntriesToShow());
+
+        this.isAnyMatchingEntries = ko.pureComputed(() => !!this.allEntries().filter(x => x.showEntry()).length);
+    }
+    
+    private computeEntriesToShow() {
+        this.allEntries().map(entry => entry.showEntry(this.shouldShowEntry(entry)));
```

Comment:
map is used to get results of map - if you ignore values then use forEach

Reply:
done